src/render/Renderer.cpp

```diff
@@ -2039,6 +2039,19 @@ bool CHyprRenderer::applyMonitorRule(CMonitor* pMonitor, SMonitorRule* pMonitorR
 
     wlr_output_set_scale(pMonitor->output, pMonitor->scale);
 
+    // Set scale for all surfaces on this monitor, needed for some clients
+    for (const auto& window : g_pCompositor->m_vWindows) {
+        if (window->m_iMonitorID == pMonitor->ID) {
+            wlr_surface_for_each_surface(
```

Comment:
you can use the updatesurfacescaletransform iirc function in CWindow (I dont remember the exact name) 

worth noting its not const so you'll have to de-const the iterator